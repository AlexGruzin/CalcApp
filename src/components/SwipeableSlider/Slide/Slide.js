import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './Slide.styles.less';

const Slide = ({ className, image, title, description }) => (
  <div className={classNames(className, 'slide')}>
    {image && (
      <div
        className="slide__image"
        style={{
          backgroundImage: `url('/images/${image}')`,
        }}
      />
    )}
    <h4 className="slide__title">{title}</h4>
    <p className="slide__description">{description}</p>
  </div>
);

Slide.propTypes = {
  className: PropTypes.string,
  image: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
};

Slide.defaultProps = {
  className: '',
  image: '',
  title: '',
  description: '',
};

export default Slide;
