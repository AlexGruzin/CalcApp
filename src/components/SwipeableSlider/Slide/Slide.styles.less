@import "~styles/generic.less";

.slide {
  display: flex;
  margin: 0;
  flex-direction: column;
  align-items: center;
  height: 408px;
  overflow: hidden;
  position: relative;
  width: 100%;
}

.slide__image {
  margin: 0;
  width: 100%;
  height: 230px;
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

.slide__title {
  color: var(--white);
  font-family: "MontserratBold", sans-serif;
  font-size: 21px;
  margin-top: var(--space-x10);
  margin-bottom: 0;
}

.slide__description {
  .montserrat-medium();
  color: var(--white);
  max-width: 550px;
  text-align: center;
}