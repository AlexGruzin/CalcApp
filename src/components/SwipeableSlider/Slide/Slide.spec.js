import React from 'react';
import { waitFor, render } from '@testing-library/react';

import Slide from './Slide';

describe('Slide tests', () => {
  it('Check snapshots of Slide are equal.', () => {
    const { asFragment } = render(
      <Slide key="key" index={1} link="/link" image="/image" title="title" description="description" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
