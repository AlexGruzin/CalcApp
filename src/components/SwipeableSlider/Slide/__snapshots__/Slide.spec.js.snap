// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide tests Check snapshots of Slide are equal. 1`] = `
<DocumentFragment>
  <div
    class="slide"
  >
    <div
      class="slide__image"
      style="background-image: url(/images//image);"
    />
    <h4
      class="slide__title"
    >
      title
    </h4>
    <p
      class="slide__description"
    >
      description
    </p>
  </div>
</DocumentFragment>
`;
